tests/dag/oneof/test_oneof_recurrent_same_ancestors.py

```diff
@@ -0,0 +1,130 @@
+import typing as t
+
+from ml_pipeline_engine.base_nodes.processors import ProcessorBase
+from ml_pipeline_engine.base_nodes.processors import RecurrentProcessor
+from ml_pipeline_engine.context.dag import DAGPipelineContext
+from ml_pipeline_engine.dag_builders.annotation.marks import GenericInput
+from ml_pipeline_engine.dag_builders.annotation.marks import Input
+from ml_pipeline_engine.dag_builders.annotation.marks import InputOneOf
+from ml_pipeline_engine.dag_builders.annotation.marks import RecurrentSubGraph
+from ml_pipeline_engine.node import build_node
+from ml_pipeline_engine.node.enums import NodeTag
+from ml_pipeline_engine.types import DAGLike
+from ml_pipeline_engine.types import Recurrent
+
+
+class MainProducer(ProcessorBase):
+    tags = (NodeTag.non_async,)
+
+    def process(self, num: int) -> int:
+        return num
+
+
+class Producer(ProcessorBase):
+    tags = (NodeTag.non_async,)
+
+    def process(self, num: Input(MainProducer), additional_data: t.Any = None) -> t.Any:
+        if additional_data:
+            return additional_data + 1
+
+        return num
+
+
+class ProcessLikeFactory(RecurrentProcessor):
+    name = 'like_factory'
+    tags = (NodeTag.non_async,)
+    kind = None
+
+    def process(self, something: GenericInput(t.Type[ProcessorBase])) -> t.Union[Recurrent, int]:
+
+        if self.kind is None:
+            raise Exception
+
+        if self.kind == 'first':
+            return self.next_iteration(100_000)
+
+        if self.kind == 'second':
+            return self.next_iteration(200_000)
+
+        return something + 3
+
+
+class ProxyNodeGeneric(ProcessorBase):
+    tags = (NodeTag.non_async,)
+
+    def process(self, arg: GenericInput(t.Type[ProcessorBase])) -> t.Any:
+        return arg
+
+
+def create_proxy_recurrent_node(name_prefix: str) -> t.Type[ProxyNodeGeneric]:
+    """
+    Shortcut to create a subgraph
+    """
+
+    process_node = build_node(
+        ProcessLikeFactory,
+        node_name=f'{name_prefix}__{ProcessLikeFactory.name}',
+        something=Input(Producer),
+        atts=dict(
```

Comment:
do you think we made a typo in the name of argument "atts" or was it by design ? 
'cause previously I wrote similar code with "attrs" as argument and of course the node was created without any attributes

Reply:
Done,. Added a compatible impl